Add README (#6)
software-mansion/starkup: README.md

# starkup: the Cairo toolchain installer

Starkup helps you install all the tools you will use to develop packages in [Cairo] and write contracts for [Starknet].

## Getting started

> Run the following in your terminal, then follow the onscreen instructions.

Reviewer: ```suggestion
Run the following in your terminal and follow the onscreen instructions.
```

Reviewer: nit

```sh
curl --proto '=https' --tlsv1.2 -sSf https://sh.starkup.dev | sh
```

## Supported tools

Starkup supports the installation of the following tools:

- [Scarb] the Cairo package manager
- [Starknet Foundry] the Cairo and Starknet testing framework
- [Universal Sierra Compiler] compiler for any ever-existing Sierra version

## Architecture

> Starkup relies on [ASDF] package manager to install the latest versions of [Scarb] and [Starknet Foundry]. If you don't have [ASDF] yet, no worries! We will install it interactively, or point you to appropriate docs.

Reviewer: ```suggestion
Starkup relies on [ASDF] package manager to install the latest versions of [Scarb] and [Starknet Foundry]. If you don't have [ASDF] yet, no worries - the script can handle that as well!
```
maybe?
